src/v/security/audit/audit_log_manager.cc

```diff
@@ -533,18 +548,24 @@ ss::future<> audit_client::produce(
           [this,
            &probe,
            max_concurrency,
-           records = std::move(records)]() mutable {
+           records = std::move(records),
+           timepoint]() mutable {
               return ss::do_with(
                 std::move(records),
-                [this, &probe, max_concurrency](auto& records) mutable {
+                timepoint,
```

Comment:
nit: if you aren't going to modify `timepoint` it would generally be better to capture it rather than incur the separate heap allocation for the do_with parameters.